deploy-tool/main.go

```diff
@@ -207,25 +266,20 @@ func createBatchSigningPublicKey(
 }
 
 func createManifest(
-	k8sClient *kubernetes.Clientset,
-	terraformOutput *TerraformOutput,
+	k8sSecretsClientGetter k8scorev1.SecretsGetter,
 	dataShareProcessorName string,
 	manifestWrapper *SpecificManifestWrapper,
+	manifestWriter manifest.DataShareProcessorSpecificManifestWriter,
+	packetEncryptionKeyCSRs manifest.PacketEncryptionKeyCSRs,
 ) error {
-	if manifestExists(terraformOutput.OwnManifestBaseURL.Value, dataShareProcessorName) {
-		log.Printf("manifest for %s exists - ignoring", dataShareProcessorName)
-		return nil
-	}
-
-	k8sSecretsClient := k8sClient.CoreV1().Secrets(manifestWrapper.KubernetesNamespace)
+	k8sSecretsClient := k8sSecretsClientGetter.Secrets(manifestWrapper.KubernetesNamespace)
 
-	newBatchSigningPublicKeys := manifest.BatchSigningPublicKeys{}
 	for name, batchSigningPublicKey := range manifestWrapper.SpecificManifest.BatchSigningPublicKeys {
 		if batchSigningPublicKey.PublicKey != "" {
-			newBatchSigningPublicKeys[name] = batchSigningPublicKey
-			continue
+			// We never create keys in Terraform, so this should never happen
+			return fmt.Errorf("unexpected batch signing key in Terraform output")
```

Comment:
might be worthwhile to indicate `name` in this Errorf since it is already formatted.
```suggestion
			return fmt.Errorf("unexpected batch signing key in Terraform output for name %s", name)
```